src/Foundation/Connect/code/Managers/LogableManager.cs

```diff
@@ -0,0 +1,53 @@
+﻿//    Copyright 2020 EPAM Systems, Inc.
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+namespace Wooli.Foundation.Connect.Managers
+{
+    using System;
+
+    using Base.Models.Logging;
+    using Base.Services.Logging;
+
+    using Sitecore.Commerce.Services;
+    using Sitecore.Diagnostics;
+
+    public abstract class LogableManager
```

Comment:
May be it wolud be better to give it more generic name like BaseManager or SafeManager?

Reply:
I'll rename it to BaseManager